src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,172 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        public Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+
+        public void setItem(E item) {
+            this.item = item;
+        }
+
+        public E getItem() {
+            return item;
+        }
+
+        public Node<E> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<E> next) {
+            this.next = next;
+        }
+
+        public Node<E> getPrev() {
+            return prev;
+        }
+
+        public void setPrev(Node<E> prev) {
+            this.prev = prev;
+        }
+
+    }
+
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            head = new Node<T>(null,value,null);
+            tail = head;
+        } else if (size > 0) {
+            linkNodes(tail,new Node<T>(tail,value,null));
+        } else {
+            throw new IndexOutOfBoundsException("Failed error");
+        }
+        size++;
```

Comment:
```suggestion
        if (size == 0) {
            head = new Node<T>(null,value,null);
            tail = head;
            return;
        } 
        linkNodes(tail,new Node<T>(tail,value,null));
        size++;
```

What other option could there be?